### ISSUE TITLE
Uninstalling the module website_quote will delete a field which is also implemented in website_sale

### ISSUE THREAD

**Issue by wtaferner** [odoo/odoo#20118]
**Impacted versions:**
10.0 and above

**Steps to reproduce:**

- Go to runbot to a database having all modules installed.
- Uninstall the module ```website_quote```
- Try to access the App Website (Traceback was in 11.0, found the issue in 10.0)
Maybe you need to try to access the field somewhere else...

**Current behavior:**

<details>
  <summary>Click for Runbot Traceback</summary>

```
Erreur:
Odoo Server Error

Traceback (most recent call last):
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 911, in __get__
    value = record.env.cache.get(record, self)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/api.py", line 960, in get
    value = self._data[field][record.id][key]
KeyError: (<odoo.sql_db.Cursor object at 0x7f57d7cc70b8>, 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 911, in __get__
    value = record.env.cache.get(record, self)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/api.py", line 960, in get
    value = self._data[field][record.id][key]
KeyError: <odoo.api.Environment object at 0x7f57d7cc73c8>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/addons/website_sale/controllers/backend.py", line 49, in fetch_dashboard_data
    groupby='product_id', orderby='product_uom_qty desc', limit=5)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/models.py", line 1831, in read_group
    result = self._read_group_raw(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/models.py", line 1927, in _read_group_raw
    self._read_group_resolve_many2one_fields(fetched_data, annotated_groupbys)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/models.py", line 1947, in _read_group_resolve_many2one_fields
    data_dict = dict(m2o_records.name_get())
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/addons/product/models/product.py", line 377, in name_get
    name = variant and "%s (%s)" % (product.name, variant) or product.name
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 915, in __get__
    self.determine_value(record)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 1026, in determine_value
    self.compute_value(recs)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 982, in compute_value
    self._compute_value(records)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 975, in _compute_value
    self.compute(records)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 566, in _compute_related
    record[self.name] = other[field.name]
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/models.py", line 4659, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 915, in __get__
    self.determine_value(record)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/fields.py", line 1016, in determine_value
    record._prefetch_field(self)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/models.py", line 2568, in _prefetch_field
    result = records.read([f.name for f in fs], load='_classic_write')
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/models.py", line 2512, in read
    self._read_from_database(stored, inherited)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/models.py", line 2631, in _read_from_database
    cr.execute(query_str, params)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/sql_db.py", line 155, in wrapper
    return f(self, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/270891-11-0-349612/odoo/sql_db.py", line 232, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: column product_template.website_description does not exist
LINE 1: ...acking","product_template"."hs_code" as "hs_code","product_t...
                                                             ^
```
</details>

**Expected behavior:**

The removal of a module which is sharing a field with any other module should never delete the shared field.

@rco-odoo @Yenthe666 @nim-odoo 
May I ask you guys to verify? I guess it is something for the Framework team and it is a major issue as it will actively delete data on removal of a module having shared fields if possible based on constraints and is very dangerous to any Odoo instance out there.

OPW-776464

**Comment by Elkasitu** [comment]
@wtaferner Hi, you are right and there is data loss involved, I will talk to Raphael about it and keep you posted, thanks for letting us know.

**Comment by wtaferner** [comment]
@Yenthe666 
Framework not Website ;-)

**Comment by Yenthe666** [comment]
@wtaferner it is in a website related module and it is not a framework functionality that is broken but a logical flaw. (my five cents 😄)

**Comment by Elkasitu** [comment]
@wtaferner We've determined that the ORM does not support overriding fields among independent modules, in this case website_description is being defined in both website_sale and website_quote, which don't depend on each other. 

Therefore the problem comes from the website_quote module itself, website_description and other fields that are already defined in website_sale (or any other independent module for that matter) should be defined in a common dependency module (in this case, website_portal_sale would be a prime example since both website_sale and website_quote depend on it), however this is tricky to add as a fix to stable versions of Odoo.

I'll forward your issue to the relevant team and tag it as website.

tl;dr: website_quote issue, not framework issue

**Comment by wtaferner** [comment]
@Elkasitu 
I see, well, I actually just expect that only the last module which is using a field is going to delete this field (should be easy to check IMHO). Anything else is luxury and should be fine for 12+ ;-)

**Comment by Elkasitu** [comment]
"fixed" in rev 333a236 for 10.0, wait for forward-port for other versions
